rpcs3/Emu/Cell/Modules/cellSaveData.cpp

```diff
@@ -91,6 +93,142 @@ vm::gvar<savedata_context> g_savedata_context;
 
 std::mutex g_savedata_mutex;
 
+static std::vector<SaveDataEntry> get_save_entries(const std::string& base_dir, const std::string& prefix)
+{
+	std::vector<SaveDataEntry> save_entries;
+
+	if (base_dir.empty() || prefix.empty())
+	{
+		return save_entries;
+	}
+
+	// get the saves matching the supplied prefix
+	for (auto&& entry : fs::dir(base_dir))
+	{
+		if (!entry.is_directory || entry.name == "." || entry.name == "..")
+		{
+			continue;
+		}
+
+		entry.name = vfs::unescape(entry.name);
+
+		if (entry.name.substr(0, prefix.size()) != prefix)
+		{
+			continue;
+		}
+
+		// PSF parameters
+		const psf::registry psf = psf::load_object(fs::file(base_dir + entry.name + "/PARAM.SFO"));
+
+		if (psf.empty())
+		{
+			continue;
+		}
+
+		SaveDataEntry save_entry;
+		save_entry.dirName   = psf.at("SAVEDATA_DIRECTORY").as_string();
+		save_entry.listParam = psf.at("SAVEDATA_LIST_PARAM").as_string();
+		save_entry.title     = psf.at("TITLE").as_string();
+		save_entry.subtitle  = psf.at("SUB_TITLE").as_string();
+		save_entry.details   = psf.at("DETAIL").as_string();
+
+		for (const auto entry2 : fs::dir(base_dir + entry.name))
+		{
+			save_entry.size += entry2.size;
+		}
+
+		save_entry.atime = entry.atime;
+		save_entry.mtime = entry.mtime;
+		save_entry.ctime = entry.ctime;
+		if (fs::file icon{base_dir + entry.name + "/ICON0.PNG"})
+			save_entry.iconBuf = icon.to_vector<uchar>();
+		save_entry.isNew = false;
+		save_entries.emplace_back(save_entry);
+	}
+
+	return save_entries;
+}
+
+static error_code select_and_delete(ppu_thread& ppu)
+{
+	std::unique_lock lock(g_savedata_mutex, std::try_to_lock);
+
+	if (!lock)
+	{
+		return CELL_SAVEDATA_ERROR_BUSY;
+	}
+
+	const std::string base_dir = vfs::get(fmt::format("/dev_hdd0/home/%08u/savedata/", Emu.GetUsrId()));
+
+	auto save_entries = get_save_entries(base_dir, Emu.GetTitleID());
+
+	s32 selected = -1;
+	s32 focused  = -1;
+
+	while (true)
+	{
+		// Yield
+		lv2_obj::sleep(ppu);
+
+		// Display Save Data List asynchronously in the GUI thread.
+		selected = Emu.GetCallbacks().get_save_dialog()->ShowSaveDataList(save_entries, focused, SAVEDATA_OP_LIST_DELETE, vm::null);
+
+		// Reschedule / user canceled
+		if (ppu.check_state() || selected == -2)
```

Comment:
If check_state() evaluates to true it should return `0`, CELL_CANCEL doesnt log the intended message.

Reply:
That function doesnt return CELL_OK

Reply:
`0` isnt CELL_OK in this context

Reply:
see savedata_op.

Reply:
Let's say it isnt about the value of the returned value but rather about its type, cellNotAnError (decltype(CELL_OK) isnt appropriate here.

Reply:
why wasn't this commented then

Reply:
Converstaion from anything not CellNotAnError to error_code triggers a log message (at Emu.Stop()), you can add a comment about it.